R/dependencies.R

```diff
@@ -0,0 +1,55 @@
+validate_deps <- function() {
+  path <- fs::path("dependencies.R")
+  content <- readLines(path, skipNul = TRUE, warn = FALSE)
+  stopifnot(
+    grepl(
+      pattern = "# This file allows packrat (used by rsconnect during deployment) to pick up dependencies.", # nolint
+      x = content[1],
+      fixed = TRUE
+    ) &
+      all(grepl("library(.*)", content[2:length(content)]))
+  )
+}
+
+read_dependencies <- function(deps = NULL, verbose = TRUE) {
+  if (fs::file_exists("dependencies.R")) {
+    if (verbose) cli::cli_alert_info("Updating existing 'dependencies.R'.")
+    deps <- c(deps, renv::dependencies("dependencies.R")$Package)
+  } else if (fs::dir_exists("app")) {
+    if (verbose) cli::cli_alert_info("Generating 'dependencies.R' based on 'app' directory.")
+    deps <- c(deps, renv::dependencies("app")$Package)
+  }
+  return(deps)
+}
+
+write_file_dependencies <- function(deps) {
```

Comment:
This would break the code because we actually have two functions and used `write_file_dependencies` on purpose -  to discern between them"

```
# dependencies.R
# low-level helper function, designed to only work with accessing and writing to a file 'dependency.R'

write_file_dependencies <- function(deps) {
  deps <- sort(unique(deps))
  deps <- purrr::map_chr(deps, function(name) glue::glue("library({name})"))
  deps <- c(
    "# This file allows packrat (used by rsconnect during deployment) to pick up dependencies.",
    deps
  )
  writeLines(deps, "dependencies.R")
}
```

```
# init.R
# high-level helper function that reads existing deps, adds rhino and writes them to file

write_dependencies <- function() {
  deps <- c("rhino", read_dependencies())
  write_file_dependencies(deps)
}
```

However, I still agree with your point that naming them like this still isn't very user-friendly. We should probably pick other names for those two functions

Reply:
We don't need the `write_dependencies()` function in `init.R`, we can simply inline it (see my comment).

Reply:
I see! OK, fixed it